### PR TITLE
PXCHARTJS-1 Setup plugin register, preserve raw script for execution …

### PR DESCRIPTION
**Pull request by vivienwong**
…without additional quotation marks

### CODE REVIEW

File: services/ChartJsUtils.cfc [hunk 1]
```diff
@@ -123,6 +124,12 @@ component {
 			config.options.scales[ scale.id ] = scaleConfig;
 		} );
 
+		// Plugins
+		var plugins = "";
+		chart.getPlugins().each( function( plugin, index ) {
```
Comment (DominicWatson) [inline services/ChartJsUtils.cfc:129]: Is this not just: 

```cfc
var plugins = ArrayToList( chart.getPlugins() );
```

?

Or, evn more concise:

```cfc
config.plugins = [ ArrayToList( chart.getPlugins() ) ];
```

The placeholder, etc. is confusing to me - why do we need to serialize first and then replace the placeholder string? (line 171 below)

File: services/ChartJsUtils.cfc [hunk 2]
```diff
@@ -159,7 +166,17 @@ component {
...
 		config = chart.applyTypeConfig( config );
 
-		return SerializeJson( config );
+		var serializedConfig = SerializeJson( config );
+
+		serializedConfig = replace( serializedConfig, '"PLUGINS_PLACEHOLDER"', plugins );
+
+		var rawScripts = chart.getRawScripts();
+
+		for ( var id in rawScripts ) {
```
Comment (DominicWatson) [inline services/ChartJsUtils.cfc:175]: Its not clear to me what is happening here and how this is used. Can you update the documentation in the README.md file with a guide as to how a developer would use this?

**Review by DominicWatson — CHANGES_REQUESTED** [review]

**Comment by DominicWatson** [comment]
Thanks for this @vivienwong - could you take a look at the review comments and feedback? TIA

File: services/ChartJsUtils.cfc (hunk 1 above)
Comment (vivienwong) [reply services/ChartJsUtils.cfc:129]: This is to preserve the plugin values without inserting additional double quotes (which will be the case if the values were added directly before serialization)

The intended outcome is 
```
new Chart(ctx, {
    plugins: [plugin, plugin 2, plugin 3]
});
```

Comment (vivienwong) [reply services/ChartJsUtils.cfc:129]: I've also added the usage in README docs

File: services/ChartJsUtils.cfc (hunk 2 above)
Comment (vivienwong) [reply services/ChartJsUtils.cfc:175]: Similar to the above, the purpose is to preserve the Javascript without being serialized. I've added documentation in the README file

**Pull request merged**